benchmark/single-source/KeyPathPerformanceTests.swift

```diff
@@ -0,0 +1,1009 @@
+    element26: 26,
+    element27: 27,
+    element28: 28,
+    element29: 29,
+    element30: 30,
+    element31: 31,
+    element32: 32,
+    element33: 33,
+    element34: 34,
+    element35: 35,
+    element36: 36,
+    element37: 37,
+    element38: 38,
+    element39: 39,
+    element40: 40,
+    element41: 41,
+    element42: 42,
+    element43: 43,
+    element44: 44,
+    element45: 45,
+    element46: 46,
+    element47: 47,
+    element48: 48,
+    element49: 49,
+    element50: 50,
+    element51: 51,
+    element52: 52,
+    element53: 53,
+    element54: 54,
+    element55: 55,
+    element56: 56,
+    element57: 57,
+    element58: 58,
+    element59: 59,
+    element60: 60,
+    element61: 61,
+    element62: 62,
+    element63: 63,
+    element64: 64,
+    element65: 65,
+    element66: 66,
+    element67: 67,
+    element68: 68,
+    element69: 69,
+    element70: 70,
+    element71: 71,
+    element72: 72,
+    element73: 73,
+    element74: 74,
+    element75: 75,
+    element76: 76,
+    element77: 77,
+    element78: 78,
+    element79: 79,
+    element80: 80,
+    element81: 81,
+    element82: 82,
+    element83: 83,
+    element84: 84,
+    element85: 85,
+    element86: 86,
+    element87: 87,
+    element88: 88,
+    element89: 89,
+    element90: 90,
+    element91: 91,
+    element92: 92,
+    element93: 93,
+    element94: 94,
+    element95: 95,
+    element96: 96,
+    element97: 97,
+    element98: 98,
+    element99: 99
+  )
+}
+
+func returnKeyPathArray() -> [AnyKeyPath] {
+  var arrayOfKeyPaths = [kp0, kp1, kp2, kp3, kp4, kp5, kp6, kp7, kp8, kp9]
+  arrayOfKeyPaths.append(contentsOf: [kp10, kp11, kp12, kp13, kp14, kp15, kp16, kp17, kp18, kp19])
+  arrayOfKeyPaths.append(contentsOf: [kp20, kp21, kp22, kp23, kp24, kp25, kp26, kp27, kp28, kp29])
+  arrayOfKeyPaths.append(contentsOf: [kp30, kp31, kp32, kp33, kp34, kp35, kp36, kp37, kp38, kp39])
+  arrayOfKeyPaths.append(contentsOf: [kp40, kp41, kp42, kp43, kp44, kp45, kp46, kp47, kp48, kp49])
+  arrayOfKeyPaths.append(contentsOf: [kp50, kp51, kp52, kp53, kp54, kp55, kp56, kp57, kp58, kp59])
+  arrayOfKeyPaths.append(contentsOf: [kp60, kp61, kp62, kp63, kp64, kp65, kp66, kp67, kp68, kp69])
+  arrayOfKeyPaths.append(contentsOf: [kp70, kp71, kp72, kp73, kp74, kp75, kp76, kp77, kp78, kp79])
+  arrayOfKeyPaths.append(contentsOf: [kp80, kp81, kp82, kp83, kp84, kp85, kp86, kp87, kp88, kp89])
+  arrayOfKeyPaths.append(contentsOf: [kp90, kp91, kp92, kp93, kp94, kp95, kp96, kp97, kp98, kp99])
+  return arrayOfKeyPaths
+}
+
+@inline(never)
+public func run_KeyPathNestedStructs(n: Int) {
+  var sum = 0
+  var index = 0
+  let iterationMultipier = 200
+  for _ in 1 ... iterationMultipier * n {
+    let element = FixedSizeArrayHolder.shared.mainArrayForNestedStructs[index]
+    sum += element[keyPath: appendedKeyPath]
+    index = (index + 1) % FixedSizeArrayHolder.shared.mainArrayForNestedStructs.count
+  }
+  assert(sum == iterationMultipier * n * expectedIntForNestedStructs)
```

Comment:
Use `check` instead of `assert`.
`assert` is a no-op in optimized builds.

Reply:
I've made this revision. 
(I do see some uses of `assert` in other files in `single-source`; I think these ought to be cleaned up if needed in another PR.)